src/DebugPoints/DebugPointManager.class.st

```diff
@@ -0,0 +1,374 @@
+	dp install.
+	DebugPoint add: dp.
+
+	"adding behaviors"
+	aListOfBehaviorClasses do: [ :bhc | dp addBehavior: bhc new ].
+
+	self notifyDebugPointAdded: dp.
+
+	^ dp
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass On: aNode [
+
+	^self installNew: aDebugPointClass On: aNode withBehaviors: {  }.
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass On: aNode forObject: anObject [
+
+	^self installNew: aDebugPointClass On: aNode forObject: anObject withBehaviors: {  }.
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass On: aNode forObject: anObject withBehaviors: aListOfBehaviorClasses [
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	| dp |
+	dp := aDebugPointClass new
+		      node: aNode;
+		      targetInstance: anObject;
+		      yourself.
+
+	^ self installDebugPoint: dp withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass On: aNode withBehaviors: aListOfBehaviorClasses [
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	| dp |
+	dp := aDebugPointClass new
+		      node: aNode;
+		      yourself.
+
+	^ self
+		  installDebugPoint: dp
+		  withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass forObject: anObject onVariableAccessNamed: aSlotNameSymbol [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  forObject: anObject
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #all
+		  withBehaviors: {  }
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass forObject: anObject onVariableAccessNamed: aSlotNameSymbol withBehaviors: aListOfBehaviorClasses [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  forObject: anObject
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: #all
+		  withBehaviors: aListOfBehaviorClasses
+]
+
+{ #category : 'API' }
+DebugPointManager class >> installNew: aDebugPointClass forObject: anObject onVariableNamed: aSlotNameSymbol accessStrategy: anAccessStrategySymbol [
+	"access strategy can be #all or #read or #write"
+
+	"this should be the single point of installation for debug points, removal is in the remove method of debug point instances"
+
+	^ self
+		  installNew: aDebugPointClass
+		  forObject: anObject
+		  onVariableNamed: aSlotNameSymbol
+		  accessStrategy: anAccessStrategySymbol
+		  withBehaviors: {  }
+]
```

Comment:
We can install a debug point without behaviors?
What should it do in this case?

Reply:
Well, if it is a basic debug point: nothing. If it is a breakpoint or a watchpoint, it will just be a basic breakpoint or watchpoint.

Anyway, this is not particularly bad. We may want to create a debug point and configure it later